### PR TITLE
Send a widget ID with Shift-F6 for automated testing

### PR DESCRIPTION
**Pull request by shundhammer**
https://hackweek.suse.com/16/projects/make-yast-testing-independent-of-keyboard-shortcuts

and

https://fate.suse.com/324098

This is the libyui part of that. This is just some common infrastructure for both the Qt and the NCurses UI to use.

### CODE REVIEW

File: package/libyui.spec [hunk 1]
```diff
@@ -32,7 +32,7 @@ BuildRequires:  cmake >= 2.8
 BuildRequires:  gcc-c++
 BuildRequires:  pkg-config
 
-Url:            http://github.com/libyui/
+Url:            http://github.com/rwx788/
```
Comment (jreidinger) [inline package/libyui.spec]: this looks wrong

File: package/libyui.spec [hunk 2]
```diff
@@ -52,7 +52,7 @@ Provides:       yast2-libyui = 2.42.0
 Obsoletes:      yast2-libyui < 2.42.0
 Requires:       yui_backend = %{so_version}
 
-Url:            http://github.com/libyui/
+Url:            http://github.com/rwx788/
```
Comment (jreidinger) [inline package/libyui.spec]: and same here

File: package/libyui.spec [hunk 3]
```diff
@@ -77,7 +76,7 @@ Requires:       glibc-devel
 Requires:       libstdc++-devel
 Requires:       %{bin_name} = %{version}
 
-Url:            http://github.com/libyui/
+Url:            http://github.com/rwx788/
```
Comment (jreidinger) [inline package/libyui.spec]: and here

File: src/YDialogSpy.cc [hunk 4]
```diff
@@ -418,9 +419,8 @@ void YDialogSpyPrivate::refreshProperties()
 
     auto widget = selectedWidget();
     if (!widget) return;
-
+    YItemCollection items;
 	auto propSet = widget->propertySet();
-	YItemCollection items;
```
Comment (jreidinger) [inline src/YDialogSpy.cc]: what is goal of this change?

File: src/YWidget.cc [hunk 5]
```diff
@@ -472,6 +474,7 @@ YWidget::getProperty( const std::string & propertyName )
     if ( propertyName == YUIProperty_VWeight		) return YPropertyValue( weight( YD_VERT  ) );
     if ( propertyName == YUIProperty_HStretch		) return YPropertyValue( stretchable( YD_HORIZ ) );
     if ( propertyName == YUIProperty_VStretch		) return YPropertyValue( stretchable( YD_VERT  ) );
+    if ( propertyName == YUIProperty_ID	&& this->hasId() ) return YPropertyValue(this->id()->toString());
```
Comment (jreidinger) [inline src/YWidget.cc]: can it happen that widget does not have ID? Other properties does not have this condition.

File: VERSION.cmake [hunk 6]
```diff
@@ -1,6 +1,6 @@
-SET( VERSION_MAJOR "3")
-SET( VERSION_MINOR "3" )
-SET( VERSION_PATCH "3" )
+SET( VERSION_MAJOR "4")
+SET( VERSION_MINOR "0" )
+SET( VERSION_PATCH "0" )
```
Comment (jreidinger) [inline VERSION.cmake]: does we need to increase so minor or not? this linkings in C++ is still a bit tricky for me, especially when new method is added.

File: package/libyui.spec (hunk 1 above)
Comment (shundhammer) [reply package/libyui.spec]: ah, right.

File: src/YWidget.cc (hunk 5 above)
Comment (shundhammer) [reply src/YWidget.cc]: Yes, this is possible. The other properties handled on this level are just simple types like string, bool, int. The widget ID is a pointer to an object which might be 0; a widget might or might not have an ID. This is why this needs to be checked. hasId() simply checks of this->priv->_id is 0.

File: src/YDialogSpy.cc (hunk 4 above)
Comment (rwx788) [reply src/YDialogSpy.cc]: This is unintended changes, thanks for spotting.

File: VERSION.cmake (hunk 6 above)
Comment (shundhammer) [reply VERSION.cmake]: The changes are still binary compatible: Just internal behaviour was changed, and there is just one more (and non-virtual) new method in the YUI class. Not sure if we want to bump that .so number every time for that.

**Comment by shundhammer** [comment]
Now that the whitespace differences have become so overwhelming in that one function with the latest commit, please append this to the URL to suppress whitespace changes:
  `?w=1`

**Comment by shundhammer** [comment]
After discussing this with @lslezak, now using less aggressive versioning: Bumped the minor version of both package and library.

**Pull request merged**